### DISCUSSION TITLE
Invalidating cache of other requests

### DISCUSSION THREAD

**Discussion by niekbr** [saloonphp/saloon#68] [Q&A]
I am building a SDK for an API and have a practical question about the caching plugin.

I have (among others) the following requests:
- `ListRecords`
- `GetRecord`
- `UpdateRecord`

If the first 2 requests are cached and I update one of the records (id 1), the cache of the first 2 requests will now be outdated. How can I practically implement the invalidation of the cache of the `ListRecords()` request and the `GetRecord(1)` request after a `UpdateRecord(1)` request?

**Comment by Sammyjo20** [comment]
You should use a custom cache key for the requests. If you don’t provide one, it will use the name of the request and the headers passed to generate a key. When you have the same requests with different arguments, you should make a custom key with that ID in the key, to make it unique. 

You could also share the same key across both show/list routes so the cache is invalidated everywhere. 

See here for more: https://docs.saloon.dev/next-steps/caching#custom-cache-keys

**Reply by justsanjit** [reply]
@Sammyjo20 Thanks for such an awesome package.

I am encountering similar problem. I am using cache plugin which caches multiple pages of an endpoint.  Once I hit Create/Update endpoint, I will have to invalidate the cache for paginated endpoints. 

Is there any way to invalidate the case without sending the request? @niekbr  What was your solution?

**Reply by niekbr** [reply]
In the end, I decided not to implement caching for this resource, since there was a chance the data would be changed outside of my application.

A custom key would have worked for `CreateRecord`/`UpdateRecord`/`DeleteRecord` <=> `GetRecord`, but invalidating the `ListRecords` at the same time would require invalidating 2 custom cache keys, which is not supported at the moment as far as I know.

If you find a solution, let me know; I'm curious!

**Reply by justsanjit** [reply]
As a temporary solution, I created multiple custom cache drivers.  For example I created `file_wp_categories` and `file_wp_authors` and few more. When I need to invalidate categories cache after running create / update category, I just flush the cache driver with 

```
Cache::store('file_wp_categories')->flush();
```


I know this is not the best solution but it's working for now until I find some better alternative.
